tests/Unit/FeedXMLGenerationTest.php

```diff
@@ -0,0 +1,405 @@
+	}
+
+	/**
+	 * @group feed
+	 */
+	public function testProductIdXML() {
+		$id_method  = $this->getProductsXmlFeedAttributeMethod( 'g:id' );
+		$product    = \WC_Helper_Product::create_simple_product();
+		$product_id = $product->get_id();
+		$xml        = $id_method( $product );
+		$this->assertEquals( "<g:id>{$product_id}</g:id>", $xml );
+	}
+
+	/**
+	 * @group feed
+	 */
+	public function testPropertyIdSimpleProductXML() {
+		$id_method = $this->getProductsXmlFeedAttributeMethod( 'item_group_id' );
+		$product   = \WC_Helper_Product::create_simple_product();
+		$xml       = $id_method( $product );
+		// Simple products have no parents so they don't have group id.
+		$this->assertEquals( "", $xml );
+	}
+
+	/**
+	 * @group feed
+	 */
+	public function testPropertyIdVariableProductXML() {
+		$group_id_method   = $this->getProductsXmlFeedAttributeMethod( 'item_group_id' );
+		$product           = new \WC_Product_Variable();
+		$variation_product = \WC_Helper_Product::create_variation_product( $product );
+		$child_product_id  = $variation_product->get_children()[0];
+		$child_product     = wc_get_product( $child_product_id );
+
+		$parent_product_id = $product->get_id();
+		$xml               = $group_id_method( $child_product );
+		// Item group id should be the parent product id.
+		$this->assertEquals( "<item_group_id>{$parent_product_id}</item_group_id>", $xml );
+	}
+
+	/**
+	 * @group feed
+	 */
+	public function testPropertyTitleXML() {
+		$title_method = $this->getProductsXmlFeedAttributeMethod( 'title' );
+		$product      = \WC_Helper_Product::create_simple_product();
+		$xml          = $title_method( $product );
+		// create_simple_product gives the product `Dummy Product` title.
+		$this->assertEquals( "<title><![CDATA[Dummy Product]]></title>", $xml );
+	}
+
+	/**
+	 * @group feed
+	 */
+	public function testPropertyProductTypeXML() {
+		$product_type_method = $this->getProductsXmlFeedAttributeMethod( 'g:product_type' );
+		$product             = \WC_Helper_Product::create_simple_product();
+		$xml                 = $product_type_method( $product );
+		// create_simple_product gives the product 'Uncategorized' type.
+		$this->assertEquals( "<g:product_type>Uncategorized</g:product_type>", $xml );
+	}
+
+	/**
+	 * @group feed
+	 */
+	public function testPropertyProductTypeVariableProductXML() {
+		$product_type_method  = $this->getProductsXmlFeedAttributeMethod( 'g:product_type' );
+		$product              = new \WC_Product_Variable();
+		$variation_product    = \WC_Helper_Product::create_variation_product( $product );
+		$variation_product_id = $variation_product->get_children()[0];
+		$variation_product    = wc_get_product( $variation_product_id );
+		$xml                  = $product_type_method( $variation_product );
+		// create_simple_product gives the product 'Uncategorized' type.
+		$this->markTestSkipped( "Wating for https://github.com/woocommerce/pinterest-for-woocommerce/pull/320");
```

Comment:
Can be removed now since #320 is merged.

Reply:
Done.